src/Http/Routing/src/Matching/DfaMatcherBuilder.cs

```diff
@@ -34,12 +36,26 @@ internal class DfaMatcherBuilder : MatcherBuilder
             ILoggerFactory loggerFactory,
             ParameterPolicyFactory parameterPolicyFactory,
             EndpointSelector selector,
-            IEnumerable<MatcherPolicy> policies)
+            IEnumerable<MatcherPolicy> policies,
+            bool? useLegacy30Behavior = null)
         {
             _loggerFactory = loggerFactory;
             _parameterPolicyFactory = parameterPolicyFactory;
             _selector = selector;
 
+            if (useLegacy30Behavior.HasValue)
```

Comment:
Comment that this is for unit testing. Real world will get the value from appcontext